src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,187 @@
+        init = atoms
+        initial_atoms: FrozenSet[GroundAtom] = frozenset(atoms)
+        objects = set(state)
+        trajectories = self._get_llm_based_state_action_predictions(
+            atoms, goal, state)
+        final_states = []
+        final_actions = []
+        final_states.append(initial_atoms)
+        if trajectories and trajectories[0]:
+            final_actions.append(
+                trajectories[0][1][0])  #dealing with initial that repeats
+        else:
+            dictionary = None
+        if trajectories:
+            for t in trajectories:
+                t[0].pop()
+                for counter in range(0, len(t[0])):
+                    if counter == 0:
+                        continue
+                    final_states.append(t[0][counter])
+                    final_actions.append(t[1][counter])
+            new_states = []
+            for s in final_states:
+                new_states.append(frozenset(s))
+            dictionary = dict(zip(new_states, final_actions))
+        else:
+            dictionary = None
+        ground_nsrts = [
+            ground_nsrt for nsrt in self._get_current_nsrts()
+            for ground_nsrt in utils.all_ground_nsrts(nsrt, objects)
+        ]
+
+        def policy(
+            sete: FrozenSet[GroundAtom]
+        ) -> Optional[Dict[FrozenSet[GroundAtom], _GroundNSRT]]:
+            if dictionary and sete in dictionary:
+                return dictionary[sete]
+            return None
+
+        def check_goal(atoms: FrozenSet[GroundAtom]) -> bool:
+            return goal.issubset(atoms)
+
+        def get_valid_actions(
+            atoms: FrozenSet[GroundAtom]
+        ) -> Iterator[Tuple[_GroundNSRT, float]]:
+            for op in utils.get_applicable_operators(ground_nsrts, atoms):
+                yield (op, 1.0)
+
+        def get_next_state(atoms: FrozenSet[GroundAtom],
+                           ground_nsrt: _GroundNSRT) -> FrozenSet[GroundAtom]:
+            return frozenset(utils.apply_operator(ground_nsrt, set(atoms)))
+
+        heuristic = utils.create_task_planning_heuristic(
+            heuristic_name=CFG.pg3_task_planning_heuristic,
+            init_atoms=init,
+            goal=goal,
+            ground_ops=ground_nsrts,
+            predicates=self._initial_predicates,
+            objects=set(state),
+        )
+        _, action_seq = utils.run_policy_guided_astar(
+            initial_state=initial_atoms,
+            check_goal=check_goal,
+            get_valid_actions=get_valid_actions,
+            get_next_state=get_next_state,
+            heuristic=heuristic,
+            policy=policy,
+            num_rollout_steps=CFG.pg3_max_policy_guided_rollout,
+            rollout_step_cost=0)
+
+        return action_seq
+
+    def _get_llm_based_state_action_predictions(
+            self, atoms: Set[GroundAtom], goal: Set[GroundAtom],
+            state: State) -> List[Tuple[List[Any], List[Any]]]:
```

Comment:
these `Any` are suspicious -- it's rare that `Any` is the right choice, since it's the most general possible type. can't these be `Set[GroundAtom]` and `_GroundNSRT` respectively?